### DISCUSSION TITLE
Implement a preview without storing the image on the server

### DISCUSSION THREAD

**Discussion by gaeundev** [vercel/next.js#27603] [Help] [answered]
Hello ! I'm a newbie to next.js...

I want to implement a preview without saving the image on the server, but I can't find a way.
The method I tried is below, what am I doing wrong?

```
import { useRef, useState } from "react";
import Image from "next/image";
import user from "../../icon/user.png";

export default function Register() {
  const [icon, setIcon] = useState({
    file: "",
    preview: user,
  });

  const iconPreview = async (e) => {
    let reader = new FileReader();
    let file = e.target.files[0];

    if (file) {
      reader.onloadend = () => {
        setIcon({ file: file, preview: reader.result });
      };
      reader.readAsDataURL(file);
    } else {
      setIcon({
        file: "",
        preview: icon.preview,
      });
    }
  };

  return (
    <div>
      <div>
        <Image src={icon.preview} alt="profile" />
        <input
          type="file"
          accept="image/png, image/jpeg"
          onChange={iconPreview}
        />
      </div>
    </div>
  );
}

```

The first static images are fine.
![image](https://user-images.githubusercontent.com/52341210/127493084-94059304-86d8-4f7d-8767-99c144219e61.png)

And when I select an image from the file it throws an error.
![image](https://user-images.githubusercontent.com/52341210/127493325-d7fa1a5f-746d-40cf-a8b8-19d8310cb28b.png)

When using the img tag in react, I used it without any problem..
I don't know how to solve it. Please help me.

**Comment by merelinguist** [comment, marked as answer]
The error is right there - you have to provide a width and height to your `Image` component, such as

```jsx
<Image src={icon.preview} width={64} height={64} alt="" />
```

**Reply by gaeundev** [reply]
Oh, my God... I've done something stupid.
Why wasn't that even possible when I was doing it alone?
I listened to you and tried again and it works fine.
You are my savior.
Thank you very much.
